From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sebastian Reichel <sebastian.reichel@collabora.com>
Date: Thu, 12 Feb 2026 22:22:54 +0100
Subject: [PATCH] arm64: dts: rockchip: Fix DP sound-dai-cells on RK3588

The DP controller has a I2S and SPDIF interface and thus needs
one cell to specify the interface being used.

Signed-off-by: Sebastian Reichel <sebastian.reichel@collabora.com>
Link: https://gitlab.collabora.com/hardware-enablement/rockchip-3588/linux/-/commit/d2872441c3d55c2cbb9c0449a6c90e22babadb8d
---
 arch/arm64/boot/dts/rockchip/rk3588-base.dtsi  | 2 +-
 arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
index 376ad04e0786..2862079e1b06 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
@@ -1848,7 +1848,7 @@ dp0: dp@fde50000 {
 		phys = <&usbdp_phy0 PHY_TYPE_DP>;
 		power-domains = <&power RK3588_PD_VO0>;
 		resets = <&cru SRST_DP0>;
-		#sound-dai-cells = <0>;
+		#sound-dai-cells = <1>;
 		status = "disabled";
 
 		ports {
diff --git a/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi
index b251bb129cdb..9c3e1c45a283 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi
@@ -223,7 +223,7 @@ dp1: dp@fde60000 {
 		phys = <&usbdp_phy1 PHY_TYPE_DP>;
 		power-domains = <&power RK3588_PD_VO0>;
 		resets = <&cru SRST_DP1>;
-		#sound-dai-cells = <0>;
+		#sound-dai-cells = <1>;
 		status = "disabled";
 
 		ports {
